src/v2/expressions.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * A CEL expression which can be evaulated during function deployment, and
+ * resolved to a value of the generic type parameter: i.e, you can pass
+ * an Expression<number> as the value of an option that normally accepts numbers.
+ */
+export type Expression<T extends string | number | boolean> = string; // eslint-disable-line
```

Comment:
In the Functions codebase an Expression<T> is not the same thing as T. Expression<T> is a base class of Param<T> but can (eventually) do a few extra things:

1. BooleanExpressions should subclass Expression<T> and adds the ternary operator to create an Expression<V>
2. All (non-boolean?) expressions should support an equality condition to make a BooleanExpression